src/main/java/calculator/CalculatorService.java

```diff
@@ -0,0 +1,77 @@
+package calculator;
+
+import com.google.code.mathparser.MathParser;
+import com.google.code.mathparser.MathParserFactory;
+
+import javax.servlet.http.HttpSession;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static controllers.ControllerConstants.EMPTY_SYMBOL;
+import static controllers.ControllerConstants.EXPRESSION;
+import static utils.ConversionUtil.deleteSpacesAndConvertListToString;
+
+
+public class CalculatorService implements Calculator {
+
+    @Override
+    public int calculate(HttpSession session) {
+        String expression = getExpression(session);
+        MathParser mathParser = MathParserFactory.create();
+
+        return mathParser.calculate(expression)
+                .doubleValue()
+                .intValue();
+    }
+
+    private String getExpression(HttpSession session) {
```

Comment:
Why does service class know something about session??
Please avoid it calculate method should retrieve List<String> expressionList and Map<String, String> attributeValueMap.